app/Console/Commands/CacheDelegatePerformance.php

```diff
@@ -27,7 +29,51 @@ final class CacheDelegatePerformance extends Command
 
     public function handle(): void
     {
-        Rounds::allByRound(Monitor::roundNumber())
-            ->each(fn ($round) => CachePastRoundPerformanceByPublicKey::dispatch($round->round, $round->public_key)->onQueue('performance'));
+        $round = Monitor::roundNumber();
+
+        $query = Round::query()
+            ->where('round', $round)
+            ->limit(Network::delegateCount())
+            ->select([
+                'rounds.public_key',
+                DB::raw('MAX(rounds.balance) as balance'),
+            ])
+            ->join('blocks', 'blocks.generator_public_key', '=', 'rounds.public_key');
+
+        collect(range($round - 6, $round - 2))->map(function ($round): array {
+            $roundStart = (int) $round * Network::delegateCount();
+
+            return [
+                // 1st block on range
+                'min' => $roundStart,
+                // 51st block on range
+                'max' => $roundStart + Network::delegateCount() - 1,
+            ];
```

Comment:
There is a `Monitor` class that should be able to calculate this based on a round. Preferably that gets used to avoid having various calculations for the same item (which introduced the off-by-one in this case)

Reply:
@ItsANameToo the `heightRangeByRound` on the `App\Services\Monitor\Monitor` class creates ranges that start in the past round, see the logs below:

First is the result of the range using `heightRangeByRound` (example 460 to 510) and second the current implementation (510 to 560)


The one from the monitor is the correct one?

```
[2021-09-30 16:50:55] testing.INFO: array (
  0 => 
  array (
    'start' => 460,
    'end' => 510,
  ),
  1 => 
  array (
    'min' => 510,
    'max' => 560,
  ),
)  
[2021-09-30 16:50:55] testing.INFO: array (
  0 => 
  array (
    'start' => 511,
    'end' => 561,
  ),
  1 => 
  array (
    'min' => 561,
    'max' => 611,
  ),
)  
```

Reply:
@alfonsobries 

Min/max of a round can be determined with `(round - 1) * delegates + 1` up to and including `round * delegates`

Round 1 = 1-51 = (0 * 51 + 1 up to and including 1 * 51
Round 2 = 52-102 = (1 * 51 + 1) up to and including 2 * 51
...
so round 10 should be from `9 * 51 + 1 = 460` to `10 * 51 = 510` (= 51 blocks)